File: SumRootToLeafNumbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int sumNumbers(TreeNode root) {
        return DFS(root, 0);
    }
    
    public int DFS(TreeNode root, int num) {
        if (root == null)
            return 0;
        
        num = num * 10 + root.val;
        if (root.left == null && root.right == null)
            return num;
        
        return DFS(root.left, num) + DFS(root.right, num);
    }
}

File: lulu/sumRootToLeafNumbers.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def __init__(self):
        self.nums = []
    def sumNumbers(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        self.traversal(root, 0)
        return sum(self.nums)
    def traversal(self, root, cur):
        if root is None:
            return
        if root and root.left is None and root.right is None:
            #leaf
            oneNum = 10 * cur + root.val
            self.nums.append(oneNum)
        if root:
            cur = 10 * cur + root.val
            self.traversal(root.left, cur)
            self.traversal(root.right, cur)
